lightning/src/util/events.rs

```diff
@@ -67,6 +68,38 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug)]
+/// Some information provided on the closure source of the channel halting.
+pub enum ClosureReason {
+	/// Closure generated from receiving a peer error message by ChannelManager::handle_error
+	CounterpartyForceClosed {
+		/// The error is coming from the peer, there *might* be a human-readable msg
+		peer_msg: Option<String>,
+	},
+	/// Closure generated from ChannelManager::force_close_channel
+	HolderForceClosed,
+	/// Closure generated from receiving a peer's ClosingSigned message. Note the shutdown
```

Comment:
I don't think it matters what the message is? Maybe "The channel was closed after negotiating a cooperative close and we've now broadcasted the cooperative close transaction. Note the shutdown may have been initiated by us"